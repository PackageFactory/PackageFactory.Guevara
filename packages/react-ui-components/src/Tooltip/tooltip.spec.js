import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import ToolTip from './tooltip';

describe('<ToolTip/>', () => {
    let props;

    beforeEach(() => {
        props = {
            theme: {
                tooltip: 'tooltipClassName'
            },
            children: 'Foo children'
        };
    });

    it('should render correctly.', () => {
        const wrapper = shallow(<ToolTip {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<ToolTip {...props} className="fooClassName"/>);

        expect(wrapper.prop('className')).toContain('fooClassName');
    });

    it('should allow the propagation of additional props to the wrapper.', () => {
        const wrapper = shallow(<ToolTip {...props} foo="bar"/>);

        expect(wrapper.prop('foo')).toBe('bar');
    });
});
