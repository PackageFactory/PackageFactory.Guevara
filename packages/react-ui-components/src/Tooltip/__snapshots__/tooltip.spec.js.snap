// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolTip/> should render correctly. 1`] = `
<div
  className="tooltipClassName"
>
  <div />
  <div>
    Foo children
  </div>
</div>
`;
